src/lib/operators/aggregate_hash.cpp

```diff
@@ -1009,9 +1119,9 @@ void AggregateHash::write_aggregate_output(ColumnID aggregate_index) {
   constexpr bool NEEDS_NULL =
       (aggregate_function != AggregateFunction::Count && aggregate_function != AggregateFunction::CountDistinct);
 
-  if (!results.empty()) {
-    write_aggregate_values<ColumnDataType, decltype(aggregate_type), aggregate_function>(values, null_values, results);
-  } else if (_groupby_column_ids.empty()) {
+  write_aggregate_values<ColumnDataType, decltype(aggregate_type), aggregate_function>(values, null_values, results);
```

Comment:
Why remove ` if (!results.empty()) {`? `write_aggregate_values` appears to do some unnecessary reserves for the empty case (don't matter) and a skipped loop.

Reply:
If `results.empty()` then we won't do any reserves and the loop condition will be only executed once. I am very sure that we will not be able to see any performance effects of this. Given the overall size of this operator's code, deleting some lines - if only a few - just felt good. ;)